docs/4.x/plugin-directory-structure.md

```diff
@@ -0,0 +1,54 @@
+|  Model/  |    This directory includes Models. These typically use `Dao` classes and have more logic.     |
+|  Reports/  |     This directory includes [reports](/guides/custom-reports). One PHP file for each report.      |
+|  screenshots/ |     This directory includes screenshots of the plugin, which are then shown on our Matomo Marketplace.    |
+|  Updates/  |     This directory includes [migration update](/guides/updates-aka-migrations) files.      |
+|  UserSettings.php  |     Defines [user settings](/guides/plugin-settings) which are shown in the personal settings page and can be edited by each user. The setting is stored differently for each user, meaning a change here doesn't affect other users' settings.    |
+|  stylesheets/ |     This directory includes CSS or less stylesheet files.    |
+|  SystemSettings.php  |    Defines [system settings](/guides/plugin-settings) which are shown in Matomo's general settings and can be edited by a superuser.     |
+|  Tasks.php  |    Defines [scheduled tasks](/guides/scheduled-tasks) that are executed periodically similar to a cron.     |
+|  templates/ |     This directory includes twig template files.    |
+|  tests/Fixtures/  |     This directory includes all kind of fixtures for our automated tests.       |
+|  tests/Framework/Mock  |     This directory may include PHP mocks used in PHP tests.      |
+|  tests/Integration/  |     This directory includes integration [PHP tests](/guides/tests-php).       |
+|  tests/javascript/index.php  |     This file includes [JavaScript tests](/guides/jstracker-core#tests) for our JS tracker and Tag Manager.      |
+|  tests/javascript/head.php  |     This file can be used to load additional JS files in JavaScript tests for our JS tracker and Tag Manager.      |
+|  tests/UI/  |     This directory includes [UI screenshot tests](/guides/tests-ui). Each test file ends with `_spec.js`.       |
+|  tests/Unit/  |     This directory includes unit [PHP tests](/guides/tests-php).       |
+|  tests/System/  |     This directory includes system [PHP tests](/guides/tests-php).       |
+|  tracker.js  |     If such a file is present, then this code will be added to the [Matomo JavaScript tracking code](/guides/enrich-js-tracker). If a `tracker.min.js` exists then the minified version will be used.   |
+|  Tracker/RequestProcessor.php  |     This file is typically used to [hook into various events during a tracking request](/guides/tracking-requests).      |
+|  Tracker/LogTable/  |     This directory can include log tables if the plugin creates custom log tables for tracking data.      |
+|  VisitorDetails.php  |    This file is used to add [additional information to the visitor log or visitor profile](/guides/visitor-log-and-profile).     |
+|  Visualizations/  |     This directory includes [visualizations](/guides/visualizing-report-data) that show report data differently.      |
```

Comment:
We could also mention `vendor/`, which is similar to `libs`, but holds composer requirements